ompi/mpi/fortran/mpif-h/register_datarep_f.c

```diff
@@ -89,28 +90,28 @@ static int extent_intercept_fn(MPI_Datatype type_c, MPI_Aint *file_extent,
 /* Data structure passed to the intercepts (see below).  It is an OPAL
    list_item_t so that we can clean this memory up during
    MPI_FINALIZE.  */
-typedef struct intercept_extra_state {
+typedef struct ompi_intercept_extra_state {
```

Comment:
I added a check for OBJ_CLASS(name) as well as OBJ_NEW(name) outside the file where OBJ_CLASS_INSTANCE(name,,,) occurred. Now I'm getting:
150 self contained cases that could be static
17 self contained but used OBJ_CLASS_DELARATION() "extern" for some reason
133 actually span multiple files

As far as coding for this I still think adding an ```opal_``` prefix makes sense. It's what the 133 cases need, the others could in principle be switched to
    static OBJ_CLASS_INSTANCE(name,,,)
and wouldn't strictly need the ```opal_``` prefix, but it wouldn't hurt anything.